stdlib/3/asyncio/locks.pyi

```diff
@@ -1,22 +1,34 @@
-from typing import Any, Callable, Type, TypeVar, Union, Optional, Awaitable
+import sys
+from typing import Any, Callable, Iterable, Type, TypeVar, Union, Optional, Awaitable
 
 from .events import AbstractEventLoop
 from .futures import Future
 from types import TracebackType
 
 _T = TypeVar('_T')
 
-class _ContextManager:
-    def __init__(self, lock: Union[Lock, Semaphore]) -> None: ...
-    def __enter__(self) -> object: ...
-    def __exit__(self, *args: Any) -> None: ...
-
-class _ContextManagerMixin(Future[_ContextManager]):
-    # Apparently this exists to *prohibit* use as a context manager.
-    def __enter__(self) -> object: ...
-    def __exit__(self, *args: Any) -> None: ...
-    def __aenter__(self) -> Awaitable[None]: ...
-    def __aexit__(self, exc_type: Optional[Type[BaseException]], exc: Optional[BaseException], tb: Optional[TracebackType]) -> Awaitable[None]: ...
+
+if sys.version_info >= (3, 9):
+    class _ContextManagerMixin:
+        def __init__(self, lock: Union[Lock, Semaphore]) -> None: ...
+        def __aenter__(self) -> Awaitable[None]: ...
+        def __aexit__(self, exc_type: Optional[Type[BaseException]], exc: Optional[BaseException], tb: Optional[TracebackType]) -> Awaitable[None]: ...
+else:
+    class _ContextManager:
+        def __init__(self, lock: Union[Lock, Semaphore]) -> None: ...
+        def __enter__(self) -> object: ...
+        def __exit__(self, *args: Any) -> None: ...
+
+    class _ContextManagerMixin(Iterable[_ContextManager]):
```

Comment:
I guess this was so you could do `with (yield from lock)` back in the dark ages of asyncio.

Reply:
Yeah, this is what that's about.